openshift-kube-apiserver/admission/autoscaling/mixedcpus/admission_test.go

```diff
@@ -0,0 +1,199 @@
+package mixedcpus
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/kubernetes/fake"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+func fakePodLister(pod *coreapi.Pod) corev1listers.PodLister {
+	indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	if pod != nil {
+		_ = indexer.Add(pod)
+	}
+	return corev1listers.NewPodLister(indexer)
+}
+
+func getMockMixedCPUsMutation(pod *coreapi.Pod) (*mixedCPUsMutation, error) {
+	m := &mixedCPUsMutation{
+		Handler:         admission.NewHandler(admission.Create),
+		client:          &fake.Clientset{},
+		podListerSynced: func() bool { return true },
+		podLister:       fakePodLister(pod),
+	}
+	if err := m.ValidateInitialization(); err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
```

Comment:
nit: can you declare functions below where they're called?

Reply:
Sure! will do